src/esp/sim/Simulator.h

```diff
@@ -101,7 +101,12 @@ class Simulator {
    * --headless mode on linux
    */
   int gpuDevice() const {
-    CORRADE_ASSERT(context_ != nullptr, "::gpuDevice: no OpenGL context.", 0);
+    if (context_ == nullptr) {
```

Comment:
FWIW, I found it useful that `gpuDevice` can silently return 0 and do nothing for the case of `createRenderer==false`. I encountered this when running some of our python programs with `createRenderer=false`.

However, if Yili is adamant here, we can make his change as suggested. The result is just that python users who try `createRenderer==false` will have to also guard/avoid their calls to `gpuDevice`.

Either is fine with me.

Reply:
> I found it useful that gpuDevice can silently return 0 and do nothing for the case of createRenderer==false.

Good to know. May I ask in which case you find it useful when it returns silently?
I was thinking assertion (again "fail fast") would be more useful because user would immediately know she made a mistake in the code.